src/Storage/TSDatabaseView.m

```diff
@@ -35,17 +32,16 @@ + (BOOL)registerUnreadDatabaseView {
         return YES;
     }
 
-    YapDatabaseViewGrouping *viewGrouping = [YapDatabaseViewGrouping
-        withObjectBlock:^NSString *(
-            YapDatabaseReadTransaction *transaction, NSString *collection, NSString *key, id object) {
-          if ([object isKindOfClass:[TSIncomingMessage class]]) {
-              TSIncomingMessage *message = (TSIncomingMessage *)object;
-              if (message.read == NO) {
-                  return message.uniqueThreadId;
-              }
-          }
-          return nil;
-        }];
+    YapDatabaseViewGrouping *viewGrouping = [YapDatabaseViewGrouping withObjectBlock:^NSString *(
+        YapDatabaseReadTransaction *transaction, NSString *collection, NSString *key, id object) {
+        if ([object conformsToProtocol:@protocol(OWSReadTracking)]) {
+            id<OWSReadTracking> possiblyRead = (id<OWSReadTracking>)object;
```

Comment:
Correct, we currently are not handling the eventual case where Signal-Desktop might know about call messages and then mark them as read.